bundles/org.openhab.core.io.net/src/main/java/org/openhab/core/io/net/exec/ExecUtil.java

```diff
@@ -28,87 +35,101 @@
  *
  * @author Pauli Anttila - Initial contribution
  * @author Kai Kreuzer - added exception logging
+ * @author Connor Petty - replaced delimiter usage with argument array
  */
+@NonNullByDefault
 public class ExecUtil {
 
     private static Logger logger = LoggerFactory.getLogger(ExecUtil.class);
 
-    /**
-     * Use this to separate between command and parameter, and also between parameters.
-     */
-    public static final String CMD_LINE_DELIMITER = "@@";
+    private static ExecutorService executor = ThreadPoolManager.getPool("ExecUtil");
 
     /**
      * <p>
-     * Executes <code>commandLine</code>. Sometimes (especially observed on MacOS) the commandLine isn't executed
-     * properly. In that cases another exec-method is to be used. To accomplish this please use the special delimiter
-     * '<code>@@</code>'. If <code>commandLine</code> contains this delimiter it is split into a String list and the
-     * special exec-method is used.
+     * Executes <code>commandLine</code>.
      *
      * <p>
      * A possible {@link IOException} gets logged but no further processing is done.
      *
      * @param commandLine the command line to execute
      */
-    public static void executeCommandLine(String commandLine) {
-        internalExecute(commandLine);
+    public static void executeCommandLine(String... commandLine) {
+        try {
+            new ProcessBuilder(commandLine).redirectError(Redirect.DISCARD).redirectOutput(Redirect.DISCARD).start();
+        } catch (IOException e) {
+            logger.warn("Error occurred when executing commandLine '{}'", commandLine, e);
+        }
     }
 
     /**
      * <p>
-     * Executes <code>commandLine</code> and return its result. Sometimes (especially observed on MacOS) the commandLine
-     * isn't executed properly. In that cases another exec-method is to be used. To accomplish this please use the
-     * special delimiter '<code>@@</code>'. If <code>commandLine</code> contains this delimiter it is split into a
-     * String list and the special exec-method is used.
+     * Executes <code>commandLine</code> and return its result.
      *
      * <p>
      * A possible {@link IOException} gets logged but no further processing is done.
      *
+     * @param timeout the max time to wait for a process to finish, 0 to wait indefinitely
```

Comment:
The timeout is only used when it is not null:

```suggestion
     * @param timeout the max time to wait for a process to finish, null to wait indefinitely
```

Reply:
oof, forgot that one.
Fixed